pkg/types/chain_reader.go

```diff
@@ -41,8 +41,40 @@ type ChainReader interface {
 	GetLatestValue(ctx context.Context, contractName string, method string, params, returnVal any) error
 
 	// Bind will override current bindings for the same contract, if one has been set and will return an error if the
-	// contract is not known by the ChainReader, or if the address is invalid
+	// contract is not known by the ChainReader, or if the Address is invalid
 	Bind(ctx context.Context, bindings []BoundContract) error
+	// TODO UnBind?, doesn't seem like its needed?
+
+	//TODO ReBindByKey, UnBindByKey, this is needed in some places where addresses change but the contract stays the same
+	// ReBindByKey(ctx context.Context, key, address string)
+	// UnBindByKey() or UnBindByKey(ctx context.Context, key, address string)
+
+	QueryKey(ctx context.Context, keys string, queryFilter QueryFilter, limitAndSort LimitAndSort, sequenceDataType any) ([]Sequence, error)
```

Comment:
keys -> key ?

Reply:
I don't have a problem with including plural and singular, but I would prefer we name these `QuerySequenceByKey`, `QuerySequenceByKeys`, etc.

(Or of course, `QueryEventsByKey`, etc. but I assume we've already banned the word "event" due to confusion with standard blockchain terminology.)

I guess it could also just be `QueryChainByKey(s)`, or simply `QueryByKey)s)`

Reply:
My reasoning for that:  usually a key is used to narrow down a query, but the thing you're actually querying for is some subset of the set of all things that might get returned. And thing thing you're querying is usually a container holding those things, like a database or a blockchain.